File: backend/src/main/java/it/unical/inf/ea/backend/dto/basics/UserBasicDTO.java
package it.unical.inf.ea.backend.dto.basics;

import it.unical.inf.ea.backend.dto.UserImageDTO;
import it.unical.inf.ea.backend.dto.enums.UserRole;
import it.unical.inf.ea.backend.dto.enums.UserStatus;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.validator.constraints.Length;

@Data
@NoArgsConstructor
@ToString
@AllArgsConstructor
@Builder
public class UserBasicDTO {

    @NotNull
    private String id;

    @NotNull
    private String lastName;

    @NotNull
    private String firstName;

    private UserImageDTO photoProfile;

    private UserStatus status;

    private UserRole role;

}

File: frontend/app/src/main/java/com/android/frontend/dto/basic/UserBasicDTO.kt
package com.android.frontend.dto.basic

import com.android.frontend.dto.UserDTO
import com.android.frontend.dto.UserImageDTO

data class UserBasicDTO(
    var id: String,
    var lastName: String,
    var firstName: String,
    var email: String,
    var phoneNumber: String? = null,
    var photoProfile: UserImageDTO? = null,
    var status: UserDTO.UserStatus? = null,
    var role: UserDTO.UserRole? = null
)

